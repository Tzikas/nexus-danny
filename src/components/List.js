import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export const Item = styled('li')``;

export const List = ({
  className,
  emptyTemplate: Empty,
  items,
  options,
  template: Template,
}) => (
  <ul className={className}>
    {items.map((item, index) => (
      <Item key={JSON.stringify(item)}>
        <Template
          {...options}
          index={index}
          item={item}
          {...typeof item === 'object' && item}
        />
      </Item>
    ))}
    {items.length === 0 && <Empty {...options} />}
  </ul>
);

List.defaultProps = {
  className: '',
  emptyTemplate() {
    return null;
  },
  items: [],
  options: {},
  template({ item }) {
    return item;
  },
};

List.propTypes = {
  className: PropTypes.string,
  emptyTemplate: PropTypes.func,
  items: PropTypes.arrayOf(PropTypes.any),
  options: PropTypes.objectOf(PropTypes.any),
  template: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
};

export default styled(List)``;
