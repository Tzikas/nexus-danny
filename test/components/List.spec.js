import React from 'react';
import { render } from 'react-testing-library';

import { List } from 'Components/List';

describe('<List />', () => {
  it('renders by default', () => {
    const { container } = render(<List />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
