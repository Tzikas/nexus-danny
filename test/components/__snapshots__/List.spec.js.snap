// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders by default 1`] = `
<ul
  class=""
/>
`;
